moveit_planners/ompl/ompl_interface/demos/generate_state_database.cpp

```diff
@@ -118,20 +119,22 @@ void computeDB(const planning_scene::PlanningScenePtr& scene, struct GenerateSta
   scene->getCurrentStateNonConst().update();
 
   ompl_interface::OMPLInterface ompl_interface(scene->getRobotModel());
+  const planning_interface::MotionPlanRequest req;  // TODO(brycew): fill?
+  ompl_interface::ModelBasedPlanningContextPtr context = ompl_interface.getPlanningContext(scene, req);
 
   ROS_INFO_STREAM_NAMED(LOGNAME, "Generating Joint Space Constraint Approximation Database for constraint:\n"
                                      << params.constraints);
 
   ompl_interface::ConstraintApproximationConstructionResults result =
-      ompl_interface.getConstraintsLibrary().addConstraintApproximation(params.constraints, params.planning_group,
-                                                                        scene, params.construction_opts);
+      context->getConstraintsLibrary()->addConstraintApproximation(params.constraints, params.planning_group, scene,
+                                                                   params.construction_opts);
```

Comment:
Specifying the constraints, planning group and scene here again feels weird when the context could/should already know them.

Reply:
That's fair, but I can't see another way to pass those along to the constraints library without some weird state being used behind the scenes.